### PR TITLE
Update cvc4 to cvc5

### PR DESCRIPTION
**Pull request by r0qs**
Depends on https://github.com/ethereum/solidity/pull/15078

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ return either an error or the result from the solver.  A default
 ``smtchecker.js`` which exports the ``smtCallback`` function that takes 1) a
 function that takes queries and returns the solving result, and 2) a solver
 configuration object. The module ``smtsolver.js`` has a few predefined solver
-configurations, and relies on Z3, Eldarica or CVC4 being installed locally.  It
+configurations, and relies on Z3, Eldarica or CVC5 being installed locally.  It
```
Comment (blishko) [inline README.md:155]:
```suggestion
configurations, and relies on Z3, Eldarica or cvc5 being installed locally.  It
```

Just a nitpick, but that's how they want to write the name of the solver now :)

**Review by blishko — COMMENTED** [review]
Looks simple enough :)

Do we need to test it or have you already done that?

**Comment by r0qs** [comment]
> Looks simple enough :)
> 
> Do we need to test it or have you already done that?

No, I have not done that, but we could indeed add some CI tests. However, even z3 is not currently tested (see: https://github.com/ethereum/solc-js/blob/master/test/smtchecker.ts#L64). Therefore, I'm not sure if it's worth adding them in this PR. We could consider adding tests for all solvers in a subsequent PR.

**Review by blishko — APPROVED** [review]
LGTM!

**Comment by blishko** [comment]
OK, let's see if we can add some tests in another PR!

**Pull request merged**